src/main/java/inventorysetups/InventorySetupsPlugin.java

```diff
@@ -1036,12 +1038,36 @@ private boolean isNewPositionInvalid(int oldPosition, int newPosition, int size)
 
 	public List<InventorySetup> filterSetups(String textToFilter)
 	{
-		final String textToFilterLower = textToFilter.toLowerCase();
 		return inventorySetups.stream()
-			.filter(i -> i.getName().toLowerCase().contains(textToFilterLower))
+			.filter(inventorySetup -> shouldDisplaySetup(inventorySetup, textToFilter.trim().toLowerCase()))
 			.collect(Collectors.toList());
 	}
 
+	private static boolean shouldDisplaySetup(InventorySetup inventorySetup, String trimmedTextToFilterLower)
+	{
+		if (trimmedTextToFilterLower.startsWith(ITEM_SEARCH_TAG) && trimmedTextToFilterLower.length() > ITEM_SEARCH_TAG.length())
+		{
+			String itemName = trimmedTextToFilterLower.substring(ITEM_SEARCH_TAG.length()).trim();
+			// Find setups containing the given item name
+			return containerContainsItemByName(inventorySetup.getInventory(), itemName) || containerContainsItemByName(inventorySetup.getEquipment(), itemName)
```

Comment:
the order here is good, just check bolt pouch last since it's least likely to have the item. very small optimization. 

I think this order makes the most sense for short circuiting:
inventory, equipment, rune pouch, filtered items, bolt pouch.

Reply:
Sure